modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/HttpUtil.java

```diff
@@ -272,57 +130,86 @@ public static HttpResponse doPost(URL endpoint, String postBody,
+        HttpURLConnection conn = null;
         HttpResponse httpResponse;
-        if (endpoint.startsWith("http://")) {
+        try {
             URL url = new URL(endpoint);
-            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn = (HttpURLConnection) url.openConnection();
             conn.setRequestMethod("GET");
             conn.setDoOutput(true);
             conn.setReadTimeout(30000);
 
             // setting headers
-            if (headers != null && headers.size() > 0) {
-                Iterator<String> itr = headers.keySet().iterator();
-                while (itr.hasNext()) {
-                    String key = itr.next();
-                    conn.setRequestProperty(key, headers.get(key));
-                }
-            }
+            setHeadersToRequest(conn, headers);
 
-            conn.connect();
+            String response = readResponse(conn);
+            httpResponse = new HttpResponse(response, conn.getResponseCode());
+            httpResponse.setResponseMessage(conn.getResponseMessage());
+            return httpResponse;
+        } finally {
+            closeAll(conn, null, null);
+        }
+    }
 
-            // Get the response
-            StringBuilder sb = new StringBuilder();
-            BufferedReader rd = null;
-            try {
-                rd = new BufferedReader(new InputStreamReader(
-                        conn.getInputStream()));
-                String line;
-                while ((line = rd.readLine()) != null) {
-                    sb.append(line);
-                }
-                httpResponse = new HttpResponse(sb.toString(), conn.getResponseCode());
-                httpResponse.setResponseMessage(conn.getResponseMessage());
+    private static void closeAll(HttpURLConnection urlConnection, Writer writer, OutputStream out) throws IOException {
+        if (urlConnection != null) {
+            urlConnection.disconnect();
+        }
+        if (out != null) {
+            out.close();
+        }
+        if (writer != null) {
+            writer.close();
+        }
+    }
 
-            } catch (IOException ignored) {
-                rd = new BufferedReader(new InputStreamReader(
-                        conn.getErrorStream()));
-                String line;
-                while ((line = rd.readLine()) != null) {
-                    sb.append(line);
-                }
-                httpResponse = new HttpResponse(sb.toString(), conn.getResponseCode());
-                httpResponse.setResponseMessage(conn.getResponseMessage());
-            } finally {
-                if (rd != null) {
-                    rd.close();
-                }
+    private static void setHeadersToRequest(HttpURLConnection urlConnection, Map<String, String> headers) {
+        if (headers != null && headers.size() > 0) {
+            Iterator<String> itr = headers.keySet().iterator();
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/pull/357/commits/d5adfb28b2122712d0da0911c0d4921cf5bd98d9